dxw/custom-post-type-menu-highlighting
# Feature/add plugin functionality

This PR adds the core functionality for this plugin. It is based on the plugin use on the Trade Knowledge project here: https://github.com/dxw/trade-knowledge/blob/develop/wp-content/plugins/ht-cpt-menu-highlighting/index.php, updated to use standard dxw plugin structure with tests etc.

The plugin allows you to mark specific menu items as the parents of chosen custom post types or taxonomy archives, by adding a `current-menu-ancestor` class to those menu items when viewing one of those custom post types/taxonomy archives.

To test:

- Add this plugin to a site with custom post types and at least one taxonomy archive (and uses the `current-menu-ancestor` class to style parent menu items, e.g. NAO)
- Add `post-type-[custom-post-type-name]` to one of the top menu items' CSS classes in the menu editor
- Add `tax-type-[taxonomy-name]` to one of the top menu items' CSS classes in the menu editor
- When you view one of those custom post types, the relevant top menu item should now be highlighted as a parent item
- When you view a taxonomy archive for that taxonomy, the relevant top menu item should now be highlighted as a parent item

We'll probably want to extend this plugin in future so you can select what menu class it adds to the top menu items, as themes may use a variety of classes to indicate active menu items (e.g. some may use `current-page-parent`, or others), but this is specifically for use on the NAO project for now, and works for that.
